nano/node/vote_processor.cpp

```diff
@@ -15,16 +15,24 @@
 
 #include <boost/format.hpp>
 
-nano::vote_processor::vote_processor (nano::node & node_a) :
-node{ node_a },
-max_votes (node_a.flags.vote_processor_capacity),
-started{ false },
-stopped{ false },
-is_active{ false },
-thread{ [this]() {
+nano::vote_processor::vote_processor (nano::signature_checker & checker_a, nano::active_transactions & active_a, nano::node_observers & observers_a, nano::stat & stats_a, nano::node_config & config_a, nano::node_flags & flags_a, nano::logger_mt & logger_a, nano::online_reps & online_reps_a, nano::ledger & ledger_a, nano::network_params & network_params_a) :
```

Comment:
Nice job removing the `node` dependency! Should be able to remove the `node.hpp` include now too